### PR TITLE
What if my users want 'dislike' feature.

### PR DESCRIPTION
**Pull request by amrnt**

### CODE REVIEW

**Comment by kirs** [comment]
When this pull request will be merged? I look forward it.

**Comment by leehambley** [comment]
I'm just +1ing this, because I'd love to be able to `dislike` in addition to likeing, I haven't reviewed @amrnt's branch thoroughly yet, but it looks quite simple.

**Comment by leehambley** [comment]
This certainly makes more sense as `dislike` as against `unlike`. I really like this @amrnt nice one, I really hope @schneems will pull this in!

File: lib/likeable/dislike.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'digest/sha1'
+
+class Likeable::Dislike
```
Comment (schneems) [inline lib/likeable/dislike.rb]: can Likeable::Dislike be a subclass of Likeable::Like ?

File: lib/likeable/user_methods.rb [hunk 2]
```diff
@@ -4,15 +4,27 @@ module Likeable::UserMethods
...
 
-
+  # Add a like to an object
+  # if already disliked, remove the dislike and add a like.
   def like!(obj)
+    obj.remove_dislike_from(self)
     obj.add_like_from(self)
   end
 
-  def unlike!(obj)
```
Comment (schneems) [inline lib/likeable/user_methods.rb:14]: removing the `unlike!` method will break backwards compatibility. Can you alias the `unlike!` method to `cancel_like!`? That will preserve intent of the method.

**Comment by schneems** [comment]
This will be great functionality. I made two comments on code, once those are addressed, I can merge this in. Once this is merged, i'll push out a new gem.

**Comment by leehambley** [comment]
I just tried to merge it, but it wasn't clean (for me, against the latest schneems master) - I'll clean this up Friday @amrnt, unless you want to? (I'll do it anyway, if you don't post back before I hit the office tomorrow!)

File: lib/likeable/user_methods.rb (hunk 2 above)
Comment (amrnt) [inline lib/likeable/user_methods.rb:14]: Done!

File: lib/likeable/dislike.rb (hunk 1 above)
Comment (amrnt) [inline lib/likeable/dislike.rb]: Hahahaha! Actually I dont know what I was doing when I copied this class!

Comment (schneems) [inline lib/likeable/dislike.rb]: yeah, don't need to repeat the method logic in the class. Subclassing should be enough.

**Comment by amrnt** [comment]
Thanks @leehambley! I just merge it and confirm that the dislike feature is working well now!

@schneems  I tested by a dummy app I built. I wanted to write some tests, but I felt so lazy to.

**Comment by leehambley** [comment]
@amrnt I've got your latest branch, your tests don't run without Redis running, mostly related to the existing tests from @schneems setting expectations on `Likeable.redis.expects(:hset).......`, but your `remove_dislike` callback isn't stubbed. 

I'm working to make sure the tests run cleanly without Redis running, and then we'll find some way to make sure we're both credited (Is there some Git magic for co-authoring commits?) for the feature. I expect to be done by the middle of the day on Saturday.

**Comment by gdub01** [comment]
I'd love the dislike feature! Any update on when this will be merged?

**Comment by leehambley** [comment]
@gdub01 For what It's worth, I tried using this fork, found that it was too much of a mess (the tests trashed my development redis database) and I wouldn't trust it in production, I wrote this in response to needing something more correctly tested, and flexible.
- https://github.com/leehambley/opinions

**Comment by leehambley** [comment]
Also - I'm not trying to hijack this thread, but I genuinely wanted Likeable to work well, but the dislike feature was so badly written it is unusable.

**Comment by gdub01** [comment]
That's a shame. Thanks for the update though. I'll look for something else.

On Thu, May 24, 2012 at 11:31 AM, Lee Hambley <
reply@reply.github.com

**Pull request closed without merging**